runtime/dynstats.c

```diff
@@ -499,12 +601,151 @@ dynstats_createCtr(dynstats_bucket_t *b, const uchar* metric, dynstats_ctr_t **c
 	RETiRet;
 }
 
+/* not supported in older platforms - disable */
+#ifndef O_CLOEXEC
+#define O_CLOEXEC 0
+#endif
+/*
+ *	Helper function adapted from imfile of the same name - possible candidate for refactor.
+ */
+static rsRetVal ATTR_NONNULL()
+atomicWriteStateFile(const char *fn, const char *content) {
```

Comment:
`fn` sounds like `function`, how about `file_name`?

Reply:
Also, the impl doesn't look atomic, so at this point the name seems misleading.

But I think we should use atomic-persist here.

How about we write this to a temp-file (in the same dir) and then `rename` it to desired file-name?

While `open` with `truncate` and `write` is not an atomic operations, `rename` is wrt the new-name.